packages/tools/devtools/devtools-view/src/DevtoolsView.tsx

```diff
@@ -183,10 +186,40 @@ export function DevtoolsView(): React.ReactElement {
 		};
 	}, [messageRelay, setSupportedFeatures]);
 
+	// Manage the query timeout
+	React.useEffect(() => {
+		if (supportedFeatures === undefined) {
+			// If we have queried for the supported feature list but have not received
+			// a response yet, queue a timer.
+			const timeout = setTimeout(() => {
+				setQueryTimedOut(false);
+			}, queryTimeoutInMilliseconds);
+			return (): void => {
+				clearTimeout(timeout);
+			};
+		}
+
+		return (): void => {
+			clearTimeout(queryTimer.current);
+		};
+	}, [supportedFeatures, setQueryTimedOut]);
+
+	function retryQuery(): void {
+		setQueryTimedOut(false);
+		messageRelay.postMessage(getSupportedFeaturesMessage);
+	}
+
 	return (
 		<FluentProvider theme={getFluentUIThemeToUse()} style={{ height: "100%" }}>
 			{supportedFeatures === undefined ? (
-				<Waiting />
+				queryTimedOut ? (
+					<>
+						<div>Devtools not found. Timeout exceeded.</div>
+						<button onClick={retryQuery}>Search again</button>
```

Comment:
Nit: I would recommend using the FluentUI button component. And we should be sure to add alt-text and a tooltip like we do elsewhere.